braindecode/datasets/base.py

```diff
@@ -0,0 +1,62 @@
+"""
+Dataset classes.
+"""
+
+# Authors: Hubert Banville <hubert.jbanville@gmail.com>
+#          Lukas Gemein <l.gemein@gmail.com>
+#          Simon Brandt <simonbrandt@protonmail.com>
+#          David Sabbagh <dav.sabbagh@gmail.com>
+#
+# License: BSD (3-clause)
+
+from torch.utils.data import Dataset
+
+
+class BaseDataset(Dataset):
+    """
+    A base dataset.
+
+    Parameters
+    ----------
+    raw: mne.Raw
+    info: pandas.DataFrame
+        holds additional information about the raw
+    """
+    def __init__(self, raw, info, target=None):
+        self.raw = raw
+        self.info = info
+        if target is not None:
+            assert target in self.info, f"'{target}' not in info"
+        self.target = target
+
+    def __getitem__(self, index):
+        return self.raw, self.target
+
+    def __len__(self):
+        return len(self.raw)
+
+
+class WindowsDataset(BaseDataset):
+    """
+    Applies a windower to a base dataset.
```

Comment:
Maybe "Dataset wrapper around an mne.Epochs object"? There is no windower here, the data is already passed as windows.